File: driftwood-e2e/src/main/kotlin/uk/co/grahamcox/driftwood/e2e/pages/HomePageSteps.kt
package uk.co.grahamcox.driftwood.e2e.pages

import cucumber.api.java8.En
import org.springframework.beans.factory.annotation.Autowired
import uk.co.grahamcox.driftwood.e2e.browser.Browser

/**
 * Cucumber steps for the Home Page
 */
class HomePageSteps : En {
    /** The browser */
    @Autowired
    private lateinit var browser: Browser

    init {
        When("I load the home page") {
            HomePage(browser).navigateTo()
        }
    }
}

File: driftwood-e2e-old/src/homePageSteps.js
const {When, Then} = require('cucumber');
const expect = require('unexpected');
const HomePage = require('./pages/HomePage');

When('I load the home page', {timeout: 60 * 1000}, async function() {
    const homePage = this._browser.getPageModel(HomePage);
    await homePage.visit();
});

Then('the page title should be {string}', async function (title) {
    const homePage = this._browser.getPageModel(HomePage);
    const header = await homePage.getHeader();
    const pageTitle = await header.getTitle();
    expect(pageTitle, 'to equal', title);
});